drivers/watchdog/wdt_smartbond.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2022 Renesas Electronics Corporation and/or its affiliates
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(wdog_smartbond, CONFIG_WDT_LOG_LEVEL);
+
+#define DT_DRV_COMPAT renesas_smartbond_watchdog
+
+/* driver data */
+struct wdog_smartbond_data {
+	/* Reload value calculated in setup */
+	uint32_t reload_val;
+#ifdef CONFIG_WDT_SMARTBOND_NMI
+	const struct device *wdog_device;
+	wdt_callback_t callback;
+#endif
+};
+
+static struct wdog_smartbond_data wdog_smartbond_dev_data = {};
+
+#define WDOG_SMARTBOND_DATA(dev) ((struct wdog_smartbond_data *)(dev)->data)
+
+static int wdg_smartbond_setup(const struct device *dev, uint8_t options)
+{
+	ARG_UNUSED(dev);
+
+	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
+		LOG_ERR("Watchdog pause in sleep is not supported");
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static int wdg_smartbond_disable(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	if (SYS_WDOG->WATCHDOG_CTRL_REG & SYS_WDOG_WATCHDOG_CTRL_REG_NMI_RST_Msk) {
+		/* watchdog cannot be stopped once started when NMI_RST is 1 */
+		return -EPERM;
+	}
+
+	GPREG->SET_FREEZE_REG = GPREG_SET_FREEZE_REG_FRZ_SYS_WDOG_Msk;
+	return 0;
+}
+
+#ifdef CONFIG_WDT_SMARTBOND_NMI
+extern void z_NmiHandlerSet(void (*pHandler)(void));
+
+static void wdog_smartbond_nmi_isr(void)
+{
+	if (wdog_smartbond_dev_data.callback) {
+		wdog_smartbond_dev_data.callback(wdog_smartbond_dev_data.wdog_device, 0);
+	}
+}
+#endif /* CONFIG_RUNTIME_NMI */
+
+static int wdg_smartbond_install_timeout(const struct device *dev,
+					 const struct wdt_timeout_cfg *config)
+{
+	struct wdog_smartbond_data *data = WDOG_SMARTBOND_DATA(dev);
```

Comment:
```suggestion
	struct wdog_smartbond_data *data = (struct wdog_smartbond_data *)dev->data;
```
Using a custom macro to get the data pointer is not recommended anymore (even though still used in several drivers).